src/cpu/aarch64/jit_uni_i8i8_pooling.cpp

```diff
@@ -0,0 +1,954 @@
+/*******************************************************************************
+* Copyright 2017-2020 Intel Corporation
+* Copyright 2020 FUJITSU LIMITED
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+#include "cpu/aarch64/jit_uni_i8i8_pooling.hpp"
+#include <math.h>
+
+#include "common/dnnl_thread.hpp"
+#include "common/utils.hpp"
+
+#include "cpu/aarch64/jit_generator.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+static inline dim_t get_offset(
+        const memory_desc_wrapper &mdw, int n, int c, int d, int h, int w) {
+    switch (mdw.ndims()) {
+        case 3: return mdw.blk_off(n, c, w);
+        case 4: return mdw.blk_off(n, c, h, w);
+        case 5: return mdw.blk_off(n, c, d, h, w);
+        default: assert(!"Invalid tensor dimension in pooling");
+    }
+    return 0;
+}
+
+using namespace Xbyak_aarch64;
+
+using namespace dnnl::impl::utils;
+using namespace dnnl::impl::types;
+using namespace alg_kind;
+
+#define GET_OFF(field) offsetof(call_params_t, field)
+
+struct call_params_t {
+    const char *src_i8;
+    const char *dst_i8;
+    size_t kd_range;
+    size_t kh_range;
+    size_t kw_range;
+    float idivider;
+    const char *src_safe_access;
+    const char *dst_safe_access;
+};
+
+template <cpu_isa_t isa>
+struct jit_uni_i8i8_pooling_fwd_ker_t : public jit_generator {
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_uni_i8i8_pooling_fwd_ker_t)
+
+    using TReg = typename cpu_isa_traits<isa>::TReg;
+
+    VReg xreg(int idx) const { return VReg(idx); }
+    ZReg yreg(int idx) const { return ZReg(xreg(idx).getIdx()); }
+    TReg vreg(int idx) const { return TReg(xreg(idx).getIdx()); }
+    // In case of avx2 with data type i8 we need to use
+    // maskmovdqu and maskmovq instructions which has its destination hardcoded in rdi.
```

Comment:
[here and other places]
I think `maskmovdqu` related comments are not relevant here!

Reply:
Thanks for your reviews. I'll delete it.